### ISSUE TITLE
Issue with  com.adpdigital.push.PushService

### ISSUE THREAD

**Issue by keyvannorouzi72** [chabokpush/chabok-client-android#15]
Hi
I have problem with PushService 
PushService process not assigned in Manifest and it runs in default process. I have another RemoteService and the second one is like the one you have. Both services run in default process and in onConnect's method of second service it wants to cast the  android.os.BinderProxy to com.adpdigital.push.PushService and Application crashed.
can you check and set a LocalBinder for PushService ?

Please check the following link, which this issue is mentioned and you could find useful answers there.
[stackoverflow question](https://stackoverflow.com/questions/28364724/getting-java-lang-classcastexception-android-os-binderproxy-every-time-i-declar?answertab=oldest#tab-top)

**Comment by behrad** [comment]
Would you please send us your crash stack trace @keyvannorouzi72 ?

**Comment by keyvannorouzi72** [comment]
when i run another service with bindService i got this error : 
java.lang.ClassCastException: android.os.BinderProxy cannot be cast to com.adpdigital.push.PushService

**Comment by Husseinhj** [comment]
Could you give us some **snippet code** and **more information** about what is your **service using**?

> We need more information to find out and also if it's happened in certain time give us the scenario when or how it's happened

**Comment by keyvannorouzi72** [comment]
I implement chabok in application class and in another Activity i run another service and i get the error that i wrote in the first comment 
here is my snippet code : 
```
Intent intent = new Intent(CONTEXT, SepPaymentService.class);
        intent.setComponent(new ComponentName("my_package_name", SepPaymentService.class.getCanonicalName()));
        intent.setPackage("my_package_name");
        intent.setAction("ir.sep.mobilepayment.action.paymentrequest");
        


        bindService(intent, new ServiceConnection() {
            @Override
            public void onServiceConnected(ComponentName componentName, IBinder iBinder) {
                // here we get error
                IPaymentRequest paymentRequest = IPaymentRequest.Stub.asInterface(iBinder);

                try {
                    Bundle bundle = new Bundle();
                    bundle.putString(SepBundleKeys.REQ_MERCHANT_ID, "MERCHANT_ID");
                    bundle.putString(SepBundleKeys.REQ_PAYMENT_PARAMS, "SAMPLE");
                    bundle.putString(SepBundleKeys.REQ_ADDITIONAL_DATA, "SAMPLE");
                    bundle.putString(SepBundleKeys.REQ_USER_MSISDN, "MOBILE_NUMBER");
                    paymentRequest.startPaymentProcess(bundle, new IPaymentResponseHandler.Stub() {
                        @Override
                        public void onResult(Bundle bundle) {
                    
                        }
                    });
                } catch (Exception e) {
               
                }


            }


            @Override
            public void onServiceDisconnected(ComponentName componentName) {
           
            }
        }, Service.BIND_AUTO_CREATE);
```

**Comment by Husseinhj** [comment]
@keyvannorouzi72 I've tried testing your code and also the code that was attached in the link, But I couldn't get your exception. Could you create **your own scenario** with a clean android project with a Github **repository**?

**Comment by keyvannorouzi72** [comment]
we push the project to https://github.com/keyvannorouzi72/chabokWithSepMPG

**Comment by Husseinhj** [comment]
@keyvannorouzi72 thanks for your help. I'll check this as soon as possible.

**Comment by Husseinhj** [comment]
@keyvannorouzi72 
This problem has been solved in version 2.13.2
